src/libertem/api.py

```diff
@@ -602,7 +605,7 @@ def run(self, job: Union[Job, Analysis], roi=None):
             return analysis.get_results(out)
         return out
 
-    def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False):
+    def run_udf(self, dataset: DataSet, udf: UDF, roi=None, progress=False) -> BufferWrapper:
```

Comment:
Closely re-read the docstring again: `BufferWrapper` is not the correct type here.